STYLEGUIDE.md

```diff
@@ -0,0 +1,172 @@
+branching point. We use indentation levels and parentheses to highlight this
+structure, which makes the code feel more organized and understandable. For
+example, when a definition part extends beyond a line, we introduce line breaks
+at the earliest branching point, clearly displaying the tree structure of the
+definition. This allows the reader to follow the branches of the tree, and to
+visually grasp the scope of each operation and argument. Consider the following
+example about Segal types:
+
+```rzk
+#def is-segal-is-local-horn-inclusion
+  ( A : U)
+  ( is-local-horn-inclusion-A : is-local-horn-inclusion A)
+  : isSegal A
+  :=
+    \ x y z f g ->
+    projection-equiv-contractible-fibers
+      ( < {t : 2 * 2 | Λ t} -> A >)
+      ( \ k ->
+        Σ ( h : hom A (k (0_2 , 0_2)) (k (1_2 , 1_2))) ,
+          ( hom2 A
+            ( k (0_2 , 0_2)) (k (1_2 , 0_2)) (k (1_2 , 1_2))
+            ( \ t -> k (t , 0_2))
+            ( \ t -> k (1_2 , t))
+            ( h)))
+      ( second
+        ( comp-equiv
+          ( Σ ( k : < {t : 2 * 2 | Λ t} -> A >) ,
+            Σ ( h : hom A (k (0_2 , 0_2)) (k (1_2 , 1_2))) ,
+              ( hom2 A
+                ( k (0_2 , 0_2)) (k (1_2 , 0_2)) (k (1_2 , 1_2))
+                ( \ t -> k (t , 0_2))
+                ( \ t -> k (1_2 , t))
+                ( h)))
+          ( < {t : 2 * 2 | Δ² t} -> A >)
+          ( < {t : 2 * 2 | Λ t} -> A >)
+          ( inv-equiv
+            ( < {t : 2 * 2 | Δ² t} -> A >)
+            ( Σ ( k : < {t : 2 * 2 | Λ t} -> A >) ,
+              Σ ( h : hom A (k (0_2 , 0_2)) (k (1_2 , 1_2))) ,
+                ( hom2 A
+                  ( k (0_2 , 0_2)) (k (1_2 , 0_2)) (k (1_2 , 1_2))
+                  ( \t -> k (t , 0_2))
+                  ( \t -> k (1_2 , t))
+                  ( h)))
+            ( equiv-horn-restriction A))
+          ( horn-restriction A , is-local-horn-inclusion-A)))
+      ( horn A x y z f g)
+```
+
+The root here is the function `projection-equiv-contractible-fibers`. It takes
+four arguments, each starting on a fresh line and is indented an extra level
+from the root. The first argument fits neatly on one line, but the second one is
+too large. In these cases, we add a line break right after the `->`-symbol
+following the lambda-abstraction, which is the earliest branching point in this
+case. The next node is again `Σ`, with two arguments. The first one fits on a
+line, but the second does not, so we add a line break between them. This process
+is continued until the definition is complete.
+
+Note also that we use parentheses to mark the branches. The extra space after
+the opening parentheses marking a branch is there to visually emphasize the tree
+structure of the definition, and synergizes with our convention to have
+two-space indentation level increases.
+
+## Naming conventions
+
+- As a main strategy, we strive to keep a tight connection between names of
```

Comment:
Should we use enumerated list to be able to more easily refer to the items?

Reply:
Sounds like a good idea